contracts/genie-airdrop-factory/src/contract.rs

```diff
@@ -145,6 +150,25 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
     Ok(res)
 }
 
+pub fn query_campaign_statuses(
+    deps: Deps,
+    addresses: Vec<String>,
+) -> StdResult<Vec<CampaignStatus>> {
+    addresses
+        .iter()
+        .map(|addr| {
+            let res: StatusResponse = deps
+                .querier
+                .query_wasm_smart(addr, &AirDropQueryMsg::Status {})?;
+
+            return Ok(CampaignStatus {
+                address: addr.into(),
+                status: res.status,
+            });
```

Comment:
nit: prefer no `return` stmts:

```suggestion
            Ok(CampaignStatus {
                address: addr.into(),
                status: res.status,
            })
```